### PR TITLE
fix config/cache dir compatibility issues, fixes #7445

### LINKED ISSUE #7445: wrong cache directories
From a discussion post:

> I noticed that with borg 2 the cache directories do *not* contain the 
> name borg which is a bit annoying and i would prefer the old behaviour.
> 
> Borg 1.2 
> /home/dietmar/.cache/borg/0a2cb8f22adaa78a1863b028f3d1629393ef0fdd99b3b61c11d871de9eaaee8b/README
> 
> Borg 2.0.0b5 
> /home/dietmar/.cache/fa673b73240ed3b75ba947fd98787fdf8ecd17204323e4a10c2754cdcc5fd547/README

this is likely due to us switching to `platformdirs` library and the related api change.

config directories need also checking, there might be the same issue.

### PR DESCRIPTION
**Pull request by nain-F49FF806**
fixes #7445.

### CODE REVIEW

File: src/borg/testsuite/helpers.py [hunk 1]
```diff
@@ -675,6 +677,22 @@ def test_get_cache_dir(monkeypatch):
         assert get_cache_dir() == "/var/tmp"
 
 
+def test_get_cache_dir_compat(monkeypatch):
+    """test that it works the same for legacy and for non-legacy implementation"""
+    monkeypatch.delenv("BORG_CACHE_DIR", raising=False)
+    monkeypatch.delenv("BORG_BASE_DIR", raising=False)
+    monkeypatch.delenv("XDG_CACHE_HOME", raising=False)
```
[the remark this answers is not in the document]
Comment (ThomasWaldmann) [reply src/borg/testsuite/helpers.py:684]: considering the cache and config tests are quite similar, the way the env vars are set up should be done in precisely the same way.

except if:
- you find a bug or shortcoming
- there is a difference requiring it to be different

**Review by ThomasWaldmann — CHANGES_REQUESTED** [review]

File: src/borg/testsuite/helpers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nain-F49FF806) [reply src/borg/testsuite/helpers.py:684]: I think some of the config compat test code is a bit redundant. Like the two `if` blocks when there need be only one (it's the same condition twice). Is there something I am missing that needs the two?

So I simplified the implementation for cache test hoping it'd be easier to follow. While retaining the order of del and set of the environment variables. 

I would propose to change the config test to match the latter ie cache one at L680.   
Does that seem fine?

[Sending a patch in a moment]

File: src/borg/testsuite/helpers.py [hunk 2]
```diff
@@ -635,21 +635,20 @@ def test_get_config_dir(monkeypatch):
...
     if not is_darwin and not is_win32:
-        monkeypatch.delenv("BORG_CONFIG_DIR", raising=False)
-        monkeypatch.delenv("XDG_CONFIG_HOME", raising=False)
         # fails on macOS: assert '/Users/tw/Library/Application Support/borg' == '/Users/tw/.config/borg'
         # fails on win32 MSYS2 (but we do not need legacy compat there).
         assert get_config_dir(legacy=False) == get_config_dir(legacy=True)
-    if not is_darwin and not is_win32:
-        monkeypatch.setenv("XDG_CONFIG_HOME", "/var/tmp/.config1")
+        monkeypatch.setenv("XDG_CONFIG_HOME", "/var/tmp/xdg.config.d")
         # fails on macOS: assert '/Users/tw/Library/Application Support/borg' == '/var/tmp/.config1/borg'
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/helpers.py:646]: if you change the code, you also need to fix the path in the comment below in the same way.

File: src/borg/testsuite/helpers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ThomasWaldmann) [reply src/borg/testsuite/helpers.py:683]: The `if`s can be merged, sure.

Guess the code is now in good sync.

File: src/borg/testsuite/helpers.py [hunk 3]
```diff
@@ -683,13 +682,13 @@ def test_get_cache_dir_compat(monkeypatch):
     monkeypatch.delenv("BORG_BASE_DIR", raising=False)
     monkeypatch.delenv("XDG_CACHE_HOME", raising=False)
     if not is_darwin and not is_win32:
-        # Please try again on darwin and win32 and update test accordingly
+        # Please try on darwin and win32 and update comment accordingly
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/helpers.py]: This needs to be done still - both macOS and win32 could be check via github CI and some temporary commit that fails, so one sees the values and can write some similar comments as seen in the config compat test.

**Review by ThomasWaldmann — APPROVED** [review]
LGTM!

**Comment by ThomasWaldmann** [comment]
Thanks!

**Pull request merged**